### PR TITLE
Add accordion element of the design system

### LINKED ISSUE #35: Create the accordion of the design system
## Overview of the Feature Request

We would like to add the accordion element of our design system as React components and document them in Storybook. This will make it easier for developers to use and maintain consistent accordion elements across our applications.

The accordion elements would be the ones documented in the [Design System Analysis document](https://docs.google.com/document/d/1OxCQO7B-yJaMWRl8EzuflcSmIaCmVSs19aoD83PjPmA/edit#heading=h.2rshrjavenkd).

These accordion elements would be using the Design tokens as SASS variables if applicable. These variables are described in [the document](https://docs.google.com/document/d/1OxCQO7B-yJaMWRl8EzuflcSmIaCmVSs19aoD83PjPmA/edit#heading=h.byjoci5aw2)

## What kind of user is the feature intended for?

This feature is intended for developers who use React to build and test components for our web applications.

## What inspired the request?

Our design system includes an accordion element that is used consistently across our application. By creating a reusable React component for this element and documenting it in Storybook, we can provide a centralized and easy-to-use reference for developers.

Some references for the Storybook:

- [Vibe Design System](https://style.monday.com/?path=/docs/foundations-colors--page)
- [Storybook Design System](https://storybook-design-system.netlify.app/?path=/docs/colors--page)
- [More references](https://storybook.js.org/showcase)

## What existing behavior do you want changed?

We want to add a new section to Storybook that displays our accordion element as React components. This section should include documentation on how to use each component, as well as examples of how the components can be customized.

## Any brand new behavior do you want to add to Dataverse?

No, this feature request does not introduce any brand new behavior to Dataverse.

## Any open or closed issues related to this feature request?

- https://github.com/IQSS/dataverse-frontend/issues/14

- https://github.com/IQSS/dataverse-frontend/issues/30

### PR DESCRIPTION
**Pull request by ekraffmiller**
## What this PR does / why we need it:
Adds an Accordion UI element
## Which issue(s) this PR closes:

- Closes #35

## Suggestions on how to test this:
npm run storybook 

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
no
## Is there a release notes update needed for this change?:
no

### CODE REVIEW

File: src/stories/ui/accordion/Accordion.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import type { Meta, StoryObj } from '@storybook/react'
+import { Accordion } from '../../../sections/ui/accordion/Accordion'
+
+const meta: Meta<typeof Accordion> = {
```
Comment (pdurbin) [inline src/stories/ui/accordion/Accordion.stories.tsx:4]: Please add a little jsdoc like https://github.com/IQSS/dataverse-frontend/pull/55/commits/a073464ae0b2460ec7ec5f5873be0624b005b538

That way we get the nice docs from Storybook for free. 😄

**Review by pdurbin — COMMENTED** [review]
A little jsdoc, please.

File: cypress/component/ui/accordion/Accordion.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+          <Accordion.Header>Section 1</Accordion.Header>
+          <Accordion.Body>Content 1</Accordion.Body>
+        </Accordion.Item>
+        <Accordion.Item eventKey="2">
+          <Accordion.Header>Section 2</Accordion.Header>
+          <Accordion.Body>Content 2</Accordion.Body>
+        </Accordion.Item>
+      </Accordion>
+    )
+    cy.pause()
```
Comment (MellyGray) [inline cypress/component/ui/accordion/Accordion.spec.tsx:17]:
```suggestion
```

Should we remove this `cy.pause` ?

File: cypress/component/ui/accordion/Accordion.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { Accordion } from '../../../../src/sections/ui/accordion/Accordion'
+
+describe('Accordion', () => {
+  it('renders an accordion with defaultActiveKey prop and children', () => {
```
Comment (MellyGray) [inline cypress/component/ui/accordion/Accordion.spec.tsx:4]: I think we should add a test 
1. to test the functionality of show/hide on click
2. to test the alwaysOpen prop

**Review by MellyGray — COMMENTED** [review]
Looks good, I just left a few comments about the tests

File: src/stories/ui/accordion/Accordion.stories.tsx (hunk 1 above)
Comment (MellyGray) [reply src/stories/ui/accordion/Accordion.stories.tsx:4]: @ekraffmiller can you take a look at this? I think is the only thing left in the review

Comment (ekraffmiller) [reply src/stories/ui/accordion/Accordion.stories.tsx:14]: yes, I added that in my latest commit

**Comment by ekraffmiller** [comment]
Hi @MellyGray I added the tests you suggested - I had to add the ThemeProvider to the rendered component, we can remove that when we merge with the code that has cy.customMount().  Also I realized the alwaysOpen behavior wasn't working exactly as expected - in order to fix it, I had to set the defaultActiveKey with an array object, rather than a string.  It's a little strange, but that's exactly how the underlying bootstrap component works.  Let me know what you think.

File: cypress/component/ui/accordion/Accordion.spec.tsx (hunk 2 above)
Comment (ekraffmiller) [reply cypress/component/ui/accordion/Accordion.spec.tsx]: yes, I removed the pause(), I just had it there to help me debug the test

**Comment by MellyGray** [comment]
> Hi @MellyGray I added the tests you suggested - I had to add the ThemeProvider to the rendered component, we can remove that when we merge with the code that has cy.customMount(). Also I realized the alwaysOpen behavior wasn't working exactly as expected - in order to fix it, I had to set the defaultActiveKey with an array object, rather than a string. It's a little strange, but that's exactly how the underlying bootstrap component works. Let me know what you think.

Look goods to me. I saw you added the defaultActiveKey array explanation in the jsdoc so it should be easy to use the component as described there.

**Review by MellyGray — APPROVED** [review]
LGTM

**Comment by kcondon** [comment]
@ekraffmiller There merge conflicts, can you please resolve? Thanks!

**Comment by ekraffmiller** [comment]
@kcondon merged with develop

**Pull request merged**